torch/distributed/pipelining/_IR.py

```diff
@@ -875,6 +885,41 @@ def move_param_to_callee(
                     node.target,
                 )
 
+        # [aliasing] store tensor id -> list of FQNs, built from state dict
+        id_to_fqns: Dict[int, List[str]] = defaultdict(list)
+        for fqn, tensor in mod.state_dict(keep_vars=True).items():
+            id_to_fqns[id(tensor)].append(fqn)
+
+        # After moving the params to their corresponding hierarchies, we also
+        # need to move the `get_attr` nodes from the root of the graph to those
+        # hierarchies.
+        # [aliasing] use id -> fqn mapping to list out all valid FQNs
+        inputs_to_state: Dict[str, List[str]] = {}
+        for attr in attr_nodes:
+            tensor = _recursive_getattr(mod, attr.target)
+            inputs_to_state[attr.name] = [fqn for fqn in id_to_fqns[id(tensor)]]
+
+        # [aliasing] for each submodule split, assign attributes on FQNs that may be used.
+        # We determine this based on whether or not the FQN attribute parent exists.
+        # i.e. if the last submodule exists, assign the attribute.
+        added_attributes: Dict[str, List[str]] = defaultdict(list)
+        for fqn, tensor in mod.state_dict(keep_vars=True).items():
+            for name, submod in split.named_children():
```

Comment:
nit: this double-for loop may be a bit time consuming. Not a blocker to land though. Maybe we can improve it later?